lib/src/repositories/refresh_polling_queue/cancellable_operation_awaited.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:async/async.dart';
+
+/// Wrapper above CancelableOperation that allows mark operation as canceled
+/// and wait for canceled operation itself except of ignoring result.
+class CancelableOperationAwaited<T> {
+  CancelableOperationAwaited.fromFuture(Future<T> result)
+      : _operation = CancelableOperation.fromFuture(result);
+
+  final CancelableOperation<T> _operation;
+
+  bool _isCanceled = false;
+
+  bool get isCanceled => _isCanceled;
+
+  Future<void> cancel() {
+    _isCanceled = true;
+    return _operation.valueOrCancellation();
```

Comment:
I don't get it: you don't call super.cancel() at all?

Reply:
Okay, but what is the point of use this library then? valueOrCancellation() will wait for super's cancellation, isn't it?